File: lab4/lab5_python/classes.py
from random import randint
class Num:
    def __init__(self,num):
        self.num=num
    def increase(self,x):
        self.num*=x
    def reducing(self,x):
        self.num/=x
    def mod(self):
        self.num=abs(num)
class RationalNum(Num):
    def __init__(self,num,denominator):
        super().__init__(num)
        self.denominator=denominator
    def reducing(self,x):
        self.denominator*=x
    def mod(self):
        return abs(self.num/self.denominator)
class CompNum(Num):
    def __init__(self,num,im):
        super().__init__(num)
        self.im=im
    def increase(self,x):
        self.num*=x
        self.im*=x
    def mod(self):
        return (self.num**2+self.im**2)**(0.5)
    
def CreateArrRational(x):
    numbers=[]
    for i in range(x):
        n1=randint(-10,10)
        n2=randint(0,10)
        num=RationalNum(n1,n2)
        numbers.append(num)
    return numbers
def CreateArrComp(x):
    numbers=[]
    for i in range(x):
        n1=randint(-10,10)
        n2=randint(0,10)
        num=CompNum(n1,n2)
        numbers.append(num)
    return numbers
def Sum(rnum,cnum):
    s=0
    for i in range(len(rnum)):
        s+=rnum[i].mod()
    for i in range(len(cnum)):
        s+=cnum[i].mod()
    return s
def out_r(rnum):
    for i in range(len(rnum)):
        print(rnum[i].num,'/',rnum[i].denominator)
def out_compl(cnum):
    for i in range(len(cnum)):
        print(cnum[i].num,"+i*",cnum[i].im)

File: lab4/lab5c++/classes.cpp
#include "classes.h"
Num::Num(int n)
{
    this->n = n;
}
Num::Num()
{
    this->n = 0;
}
void Num::increase(int x)
{
    n *= x;
}
void Num::reducing(int x)
{
    n /= x;
}
double Num::mod()
{
    return abs(n);
}
RationalNum::RationalNum(int num, int denominator) :Num(num)
{
    this->denominator = denominator;
}
void RationalNum::reducing(int x)
{
    denominator *= x;
}
double RationalNum::mod()
{
    double x = fabs((double)n / (double)denominator);
    return x;
}
ostream& operator<<(ostream& out, const RationalNum& num)
{

    out << num.n << "/" << num.denominator << endl;
    return out;
}
CompNum::CompNum(int re, int im) :Num(re)
{
    this->im = im;
}
void CompNum::increase(int x)
{
    n *= x;
    im *= x;
}
double CompNum::mod()
{
    return sqrt(pow(n, 2) + pow(im, 2));
}
ostream& operator<<(ostream& out, const CompNum& num)
{
    out << num.n << "+i*" << num.im << endl;
    return out;
}
vector<RationalNum> CreateArrRational(int x)
{
    srand(time(NULL));
    vector<RationalNum> numbers;
    for (int i = 0;i < x;i++)
    {
        int n1 = rand() % 10 - 10;
        int n2 = rand() % 10 + 1;
        RationalNum num(n1, n2);
        numbers.push_back(num);
    }
    return numbers;
}
vector<CompNum> CreateArrComp(int x)
{
    srand(time(NULL));
    vector<CompNum> numbers;
    for (int i = 0;i < x;i++)
    {
        int n1 = rand() % 10;
        int n2 = rand() % 10;
        CompNum num(n1, n2);
        numbers.push_back(num);
    }
    return numbers;
}
double Sum(vector<RationalNum> rnum, vector<CompNum> cnum)
{
    double s = 0;
    for (int i = 0;i < rnum.size();i++)
    {
        s += rnum[i].mod();
    }
    for (int i = 0;i < cnum.size();i++)
    {
        s += cnum[i].mod();
    }
    return s;
}